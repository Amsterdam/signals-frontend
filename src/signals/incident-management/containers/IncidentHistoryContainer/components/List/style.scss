.incident-history-container-list {
  &__body-list {
    width: 100%;
  }
}
