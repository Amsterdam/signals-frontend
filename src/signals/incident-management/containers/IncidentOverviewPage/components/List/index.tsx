// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2018 - 2023 Gemeente Amsterdam
import type { FunctionComponent, ReactNode } from 'react'
import { useEffect, useState } from 'react'

import { Play } from '@amsterdam/asc-assets'
import differenceInCalendarDays from 'date-fns/differenceInCalendarDays'
import parseISO from 'date-fns/parseISO'
import { Link, useNavigate } from 'react-router-dom'

import ParentIncidentIcon from 'components/ParentIncidentIcon'
import configuration from 'shared/services/configuration/configuration'
import { formatAddress } from 'shared/services/format-address'
import {
  getListIconByKey,
  getListValueByKey,
} from 'shared/services/list-helpers/list-helpers'
import { string2date, string2time } from 'shared/services/string-parser'
import { statusList } from 'signals/incident-management/definitions'
import type {
  Definition,
  Priority,
  Status,
} from 'signals/incident-management/definitions/types'
import { INCIDENT_URL } from 'signals/incident-management/routes'
import type { IncidentList, IncidentListItem } from 'types/api/incident-list'
import onButtonPress from 'utils/on-button-press'

import {
  BaseTh,
  ContentSpan,
  StyledIcon,
  StyledList,
  Table,
  TdStyle,
  ThArea,
  ThDate,
  ThDay,
  ThId,
  ThParent,
  ThPriority,
  ThStatus,
  ThSubcategory,
  Tr,
} from './styles'
import { useIncidentManagementContext } from '../../../../context'
import { SortOptionLabels, SortOptions } from '../../contants'
import { compareSortOptions } from '../../utils'
import ThSort from '../Th'

export const getDaysOpen = (incident: IncidentListItem) => {
  const statusesWithoutDaysOpen = statusList
    .filter(({ shows_remaining_sla_days }) => !shows_remaining_sla_days)
    .map(({ key }) => key)
  const hasDaysOpen =
    incident.status && !statusesWithoutDaysOpen.includes(incident.status.state)

  const createdAtDate = parseISO(incident.created_at)

  if (!hasDaysOpen || isNaN(createdAtDate.getTime())) return '-'

  return differenceInCalendarDays(new Date(), createdAtDate)
}
interface TdProps {
  detailLink: string
  children: ReactNode
}

const Td = ({ detailLink, children, ...rest }: TdProps) => {
  return (
    <TdStyle {...rest}>
      <Link to={detailLink} tabIndex={-1}>
        <ContentSpan>{children}</ContentSpan>
      </Link>
    </TdStyle>
  )
}

const ChildIcon: FunctionComponent = () => (
  <StyledIcon aria-label="Deelmelding" data-testid="child-icon">
    <Play />
  </StyledIcon>
)

interface ListProps {
  className?: string
  incidents: IncidentList
  isLoading?: boolean
  priority: Priority[]
  stadsdeel: Definition[]
  status: Status[]
  orderingChangedAction: (ordering: string) => void
  ordering?: SortOptions
  sortingDisabled?: boolean
}

const List: FunctionComponent<ListProps> = ({
  className,
  incidents,
  isLoading = false,
  priority,
  stadsdeel,
  status,
  ordering,
  orderingChangedAction,
  sortingDisabled = false,
}) => {
  const { districts, referrer } = useIncidentManagementContext()
  const navigate = useNavigate()

  const [lastId, setLastId] = useState<number | undefined>()

  const navigateToIncident = (id: number) => {
    navigate(`../${INCIDENT_URL}/${id}`)
  }

  /**
   * This method reverses the ordering if the column is already ordered
   * otherwise it will order the new column ascending
   * @param column
   */
  const changeOrder = (column: SortOptions) => {
    if (ordering && compareSortOptions(ordering, column)) {
      if (ordering.startsWith('-')) {
        orderingChangedAction(ordering.replace('-', ''))
      } else {
        orderingChangedAction(`-${ordering}`)
      }
    } else {
      orderingChangedAction(column)
    }
  }

  useEffect(() => {
    // check if referrer is incident detail page
    if (referrer?.startsWith('/manage/incident/')) {
      const lastIncidentId = Number(referrer.replace('/manage/incident/', ''))
      setLastId(lastIncidentId)
    }
  }, [referrer])

  // reset lastId on click
  useEffect(() => {
    const handleClick = () => {
      setLastId(undefined)
    }

    window.addEventListener('click', handleClick)
    return () => window.removeEventListener('click', handleClick)
  }, [])

  return (
    <StyledList
      isLoading={isLoading}
      className={className}
      data-testid="incident-overview-list-component"
    >
      <Table cellSpacing="0">
        <thead>
          <tr>
            <ThParent />
            <ThSort
              StyledComponent={ThPriority}
              sortOption={SortOptions.PRIORITY_ASC}
              headerText={SortOptionLabels.PRIORITY}
              ordering={ordering}
              changeOrder={changeOrder}
              sortingDisabled={sortingDisabled}
            />
            <ThSort
              StyledComponent={ThId}
              sortOption={SortOptions.ID_DESC}
              headerText={SortOptionLabels.ID}
              ordering={ordering}
              changeOrder={changeOrder}
              sortingDisabled={sortingDisabled}
            />
            <ThDay $isDisabled={true}>Dag</ThDay>
            <ThSort
              StyledComponent={ThDate}
              sortOption={SortOptions.CREATED_AT_DESC}
              headerText={SortOptionLabels.DATE}
              ordering={ordering}
              changeOrder={changeOrder}
            />
            <ThSort
              StyledComponent={ThSubcategory}
              sortOption={SortOptions.SUBCATEGORY_ASC}
              headerText={SortOptionLabels.SUBCATEGORY}
              ordering={ordering}
              changeOrder={changeOrder}
              sortingDisabled={sortingDisabled}
            />
            <ThSort
              StyledComponent={ThStatus}
              sortOption={SortOptions.STATUS_ASC}
              headerText={SortOptionLabels.STATUS}
              ordering={ordering}
              changeOrder={changeOrder}
              sortingDisabled={sortingDisabled}
            />
            {configuration.featureFlags.fetchDistrictsFromBackend ? (
              <ThSort
                StyledComponent={ThArea}
                sortOption={SortOptions.DISTRICT_ASC}
                headerText={configuration.language.district}
                ordering={ordering}
                changeOrder={changeOrder}
                sortingDisabled={sortingDisabled}
              />
            ) : (
              <ThSort
                StyledComponent={ThArea}
                sortOption={SortOptions.BUROUGH_ASC}
                headerText={SortOptionLabels.BUROUGH}
                ordering={ordering}
                changeOrder={changeOrder}
                sortingDisabled={sortingDisabled}
              />
            )}
            <ThSort
              StyledComponent={BaseTh}
              sortOption={SortOptions.ADDRESS_ASC}
              headerText={SortOptionLabels.ADDRESS}
              ordering={ordering}
              changeOrder={changeOrder}
              sortingDisabled={sortingDisabled}
            />
            {configuration.featureFlags.assignSignalToEmployee && (
              <ThSort
                StyledComponent={BaseTh}
                sortOption={SortOptions.ASSIGNED_USER_EMAIL_ASC}
                headerText={SortOptionLabels.ASSIGNED_USER_EMAIL}
                ordering={ordering}
                changeOrder={changeOrder}
                sortingDisabled={sortingDisabled}
              />
            )}
          </tr>
        </thead>
        <tbody>
          {incidents.map((incident) => {
            const detailLink = `/manage/incident/${incident.id}`
            return (
              <Tr
                key={incident.id}
                tabIndex={0}
                onKeyDown={(e) => {
                  onButtonPress(e, () => navigateToIncident(incident.id))
                }}
                $lastIncident={incident.id === lastId}
              >
                <Td detailLink={detailLink} data-testid="incident-parent">
                  {incident.has_children && <ParentIncidentIcon />}
                  {incident.has_parent && <ChildIcon />}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-urgency">
                  <StyledIcon>
                    {getListIconByKey(priority, incident.priority?.priority)}
                  </StyledIcon>
                </Td>
                <Td detailLink={detailLink} data-testid="incident-id">
                  {incident.id}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-days-open">
                  {getDaysOpen(incident)}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-created-at">
                  {string2date(incident.created_at)}{' '}
                  {string2time(incident.created_at)}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-subcategory">
                  {incident.category?.sub}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-status">
                  {getListValueByKey(status, incident.status?.state)}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-area">
                  {configuration.featureFlags.fetchDistrictsFromBackend
                    ? getListValueByKey(districts, incident.location?.area_code)
                    : getListValueByKey(
                        stadsdeel,
                        incident.location?.stadsdeel
                      )}
                </Td>
                <Td detailLink={detailLink} data-testid="incident-address">
                  {incident.location?.address &&
                    formatAddress(incident.location?.address)}
                </Td>
                {configuration.featureFlags.assignSignalToEmployee && (
                  <Td
                    detailLink={detailLink}
                    data-testid="incident-assigned-user"
                  >
                    {incident.assigned_user_email}
                  </Td>
                )}
              </Tr>
            )
          })}
        </tbody>
      </Table>
    </StyledList>
  )
}

export default List
