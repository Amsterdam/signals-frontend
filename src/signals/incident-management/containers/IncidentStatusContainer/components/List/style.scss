.incident-status-container-list {
  &__body-list {
    width: 100%;
  }
}
