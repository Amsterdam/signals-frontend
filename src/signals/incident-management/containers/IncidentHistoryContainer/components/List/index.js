import React from 'react';
import PropTypes from 'prop-types';

import { string2date, string2time } from 'shared/services/string-parser/string-parser';
import './style.scss';

class List extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { incidentHistoryList } = this.props;
    return (
      <div className="incident-history-container-list">
        <div className="incident-history-container-list__body">
          <table className="incident-history-container-list__body-list" cellSpacing="0" cellPadding="0">
            <thead>
              <tr>
                <th className="">Wanneer</th>
                <th className="">Wat</th>
                <th className="">Omschrijving</th>
                <th className="">Gebruiker</th>
              </tr>
            </thead>
            <tbody>
              {incidentHistoryList.map((item) => (
                <tr key={item.identifier}>
                  <td>{string2date(item.when)} om {string2time(item.when)}</td>
                  <td>{item.action}</td>
                  <td className="pre-wrap">{item.description}</td>
                  <td>{item.who}</td>
                </tr>
              ))
              }
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

List.propTypes = {
  incidentHistoryList: PropTypes.array.isRequired,
};

List.defaultProps = {
  incidentHistoryList: []
};

export default List;
