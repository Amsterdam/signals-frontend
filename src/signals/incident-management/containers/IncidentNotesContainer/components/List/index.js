import React from 'react';
import PropTypes from 'prop-types';

import { string2date, string2time } from 'shared/services/string-parser/string-parser';
import './style.scss';

class List extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { incidentNotesList } = this.props;
    return (
      <div className="incident-notes-container-list">
        <div className="incident-notes-container-list__body">
          <table className="incident-notes-container-list__body-list" cellSpacing="0" cellPadding="0">
            <thead>
              <tr>
                <th className="">Datum</th>
                <th className="">Tijd</th>
                <th className="">Notitie</th>
                <th className="">Gebruiker</th>
              </tr>
            </thead>
            <tbody>
              {incidentNotesList.map((item) => (
                <tr key={item._links.self.href}>
                  <td>{string2date(item.created_at)}</td>
                  <td>{string2time(item.created_at)}</td>
                  <td className="pre-wrap">{item.text}</td>
                  <td>{item.created_by}</td>
                </tr>
              ))
              }
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

List.propTypes = {
  incidentNotesList: PropTypes.array.isRequired,
};

List.defaultProps = {
  incidentNotesList: []
};

export default List;
