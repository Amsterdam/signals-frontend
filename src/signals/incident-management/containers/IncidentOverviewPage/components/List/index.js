import React from 'react';
import PropTypes from 'prop-types';

import { string2date, string2time } from 'shared/services/string-parser/string-parser';
import { getListValueByKey } from 'shared/services/list-helper/list-helper';
import './style.scss';

class List extends React.Component { // eslint-disable-line react/prefer-stateless-function
  onSort = (sort) => () => {
    if (this.props.sort && this.props.sort.indexOf(sort) === 0) {
      this.props.onRequestIncidents({ sort: `-${sort}` });
    } else {
      this.props.onRequestIncidents({ sort });
    }
  }

  selectIncident = (incident) => () => {
    this.props.incidentSelected(incident);
  }

  sortClassName(sortName) {
    let className = '';
    const currentSort = this.props.sort && this.props.sort.split(',')[0];
    if (currentSort && currentSort.indexOf(sortName) > -1) {
      className = currentSort.charAt(0) === '-' ? 'sort sort-down' : 'sort sort-up';
    }
    return className;
  }

  render() {
    const { incidents, incidentsCount, priorityList, statusList, stadsdeelList } = this.props;
    return (
      <div className="list-component">
        <div className="list-component__title">Meldingen ({incidentsCount})</div>
        <div className="list-component__body">
          <table className="list-component__table" cellSpacing="0" cellPadding="0">
            <thead>
              <tr>
                <th onClick={this.onSort('id')} className={this.sortClassName('id')} >Id</th>
                <th onClick={this.onSort('created_at')} className={this.sortClassName('created_at')}>Datum en tijd</th>
                <th onClick={this.onSort('stadsdeel,-created_at')} className={this.sortClassName('stadsdeel')}>Stadsdeel</th>
                <th onClick={this.onSort('sub_category,-created_at')} className={this.sortClassName('sub_category')}>Subcategorie</th>
                <th onClick={this.onSort('status,-created_at')} className={this.sortClassName('status')}>Status</th>
                <th onClick={this.onSort('priority,-created_at')} className={this.sortClassName('priority')}>Urgentie</th>
                <th onClick={this.onSort('address,-created_at')} className={this.sortClassName('address')}>Adres</th>
              </tr>
            </thead>
            <tbody>
              {incidents.map((incident) => (
                <tr key={incident.id} onClick={this.selectIncident(incident)}>
                  <td>{incident.id}</td>
                  <td className="no-wrap">{string2date(incident.incident_date_start)} {string2time(incident.incident_date_start)}</td>
                  <td>{getListValueByKey(stadsdeelList, incident.location.stadsdeel)}</td>
                  <td>{incident.category && incident.category.sub}</td>
                  <td>{getListValueByKey(statusList, incident.status.state)}</td>
                  <td>{getListValueByKey(priorityList, incident.priority && incident.priority.priority)}</td>
                  <td>{incident.location.address_text}</td>
                </tr>
              ))
              }
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

List.propTypes = {
  incidentsCount: PropTypes.number,
  incidents: PropTypes.array.isRequired,
  priorityList: PropTypes.array.isRequired,
  statusList: PropTypes.array.isRequired,
  stadsdeelList: PropTypes.array.isRequired,

  incidentSelected: PropTypes.func.isRequired,
  onRequestIncidents: PropTypes.func.isRequired,
  sort: PropTypes.string
};

export default List;
