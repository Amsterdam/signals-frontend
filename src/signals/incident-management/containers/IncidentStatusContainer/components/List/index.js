import React from 'react';
import PropTypes from 'prop-types';

import { string2date, string2time } from 'shared/services/string-parser/string-parser';
import './style.scss';

class List extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { incidentStatusList } = this.props;
    return (
      <div className="incident-status-container-list">
        <div className="incident-status-container-list__body">
          <table className="incident-status-container-list__body-list" cellSpacing="0" cellPadding="0">
            <thead>
              <tr>
                <th className="">Datum</th>
                <th className="">Tijd</th>
                <th className="">Status</th>
                <th className="">Omschrijving</th>
                <th className="">Gebruiker</th>
              </tr>
            </thead>
            <tbody>
              {incidentStatusList.map((item) => (
                <tr key={item._links.self.href}>
                  <td>{string2date(item.created_at)}</td>
                  <td>{string2time(item.created_at)}</td>
                  <td>{item.state_display}</td>
                  <td className="pre-wrap">{item.text}</td>
                  <td>{item.user}</td>
                </tr>
              ))
              }
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

List.propTypes = {
  incidentStatusList: PropTypes.array.isRequired,
};

List.defaultProps = {
  incidentStatusList: [],
  statusList: []
};

export default List;
