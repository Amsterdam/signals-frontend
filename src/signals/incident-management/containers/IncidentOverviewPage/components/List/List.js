import React, { useCallback, useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import parseISO from 'date-fns/parseISO';
import differenceInCalendarDays from 'date-fns/differenceInCalendarDays';
import { ChevronUp, ChevronDown } from '@datapunt/asc-assets';
import { Icon } from '@datapunt/asc-ui';

import { string2date, string2time } from 'shared/services/string-parser';
import { getListValueByKey } from 'shared/services/list-helper/list-helper';
import * as types from 'shared/types';
import configuration from 'shared/services/configuration/configuration';
import { statusList } from 'signals/incident-management/definitions';

import * as S from './List.styles';
import IncidentManagementContext from '../../../../context';

const getDaysOpen = incident => {
  const statusesWithoutDaysOpen = statusList
    .filter(({ shows_remaining_sla_days }) => shows_remaining_sla_days === false)
    .map(({ key }) => key);
  const hasDaysOpen = incident.status && !statusesWithoutDaysOpen.includes(incident.status.state);
  const start = hasDaysOpen && parseISO(incident.created_at);
  return hasDaysOpen ? -differenceInCalendarDays(start, new Date()) : '-';
};

const List = ({
  className = '',
  incidents,
  isLoading = false,
  onChangeOrdering,
  priority,
  sort,
  stadsdeel,
  status,
}) => {
  const { districts, users } = useContext(IncidentManagementContext);

  const onSort = useCallback(
    newSort => () => {
      const sortIsAsc = sort?.indexOf(newSort) === 0;
      onChangeOrdering(sortIsAsc ? `-${newSort}` : newSort);
    },
    [onChangeOrdering, sort]
  );

  const renderChevron = useCallback(
    column => {
      const currentSort = sort?.split(',')[0];
      const isColumnSorted = currentSort?.indexOf(column) > -1;
      const sortDirection = currentSort?.charAt(0) === '-' ? 'down' : 'up';

      return isColumnSorted ? (
        <Icon inline size={12}>
          {sortDirection === 'up' ? <ChevronDown /> : <ChevronUp />}
        </Icon>
      ) : null;
    },
    [sort]
  );

  const getAssignedUserName = useCallback(userId => users?.find(user => user.id === userId)?.username, [users]);

  return (
    <S.List isLoading={isLoading} className={className} data-testid="incidentOverviewListComponent">
      <S.Table cellSpacing="0">
        <thead>
          <tr>
            <S.Th data-testid="sortId" onClick={onSort('id')}>
              Id {renderChevron('id')}
            </S.Th>
            <S.Th data-testid="sortDaysOpen" onClick={onSort('days_open')}>
              Dag {renderChevron('days_open')}
            </S.Th>
            <S.Th data-testid="sortCreatedAt" onClick={onSort('created_at')}>
              Datum en tijd {renderChevron('created_at')}
            </S.Th>
            {configuration.fetchDistrictsFromBackend ? (
              <S.Th data-testid="sortDistrict" onClick={onSort('district,-created_at')}>
                {configuration.language.district} {renderChevron('district')}
              </S.Th>
            ) : (
              <S.Th data-testid="sortStadsdeel" onClick={onSort('stadsdeel,-created_at')}>
                Stadsdeel {renderChevron('stadsdeel')}
              </S.Th>
            )}
            <S.Th data-testid="sortSubcategory" onClick={onSort('sub_category,-created_at')}>
              Subcategorie {renderChevron('sub_category')}
            </S.Th>
            <S.Th data-testid="sortStatus" onClick={onSort('status,-created_at')}>
              Status {renderChevron('status')}
            </S.Th>
            <S.Th data-testid="sortPriority" onClick={onSort('priority,-created_at')}>
              Urgentie {renderChevron('priority')}
            </S.Th>
            <S.Th data-testid="sortAddress" onClick={onSort('address,-created_at')}>
              Adres {renderChevron('address')}
            </S.Th>
            {users && <th data-testid="sortAssigedUserId">Toegewezen aan</th>}
          </tr>
        </thead>
        <tbody>
          {incidents.map(incident => {
            const detailLink = `/manage/incident/${incident.id}`;
            return (
              <tr key={incident.id}>
                <S.Td>
                  <Link to={detailLink}>{incident.id}</Link>
                </S.Td>
                <S.Td data-testid="incidentDaysOpen">
                  <Link to={detailLink}>{getDaysOpen(incident)}</Link>
                </S.Td>
                <S.Td noWrap>
                  <Link to={detailLink}>
                    {string2date(incident.created_at)} {string2time(incident.created_at)}
                  </Link>
                </S.Td>
                <S.Td>
                  <Link to={detailLink}>
                    {configuration.fetchDistrictsFromBackend
                      ? getListValueByKey(districts, incident.location && incident.location.area_code)
                      : getListValueByKey(stadsdeel, incident.location && incident.location.stadsdeel)}
                  </Link>
                </S.Td>
                <S.Td>
                  <Link to={detailLink}>{incident.category && incident.category.sub}</Link>
                </S.Td>
                <S.Td>
                  <Link to={detailLink}>{getListValueByKey(status, incident.status && incident.status.state)}</Link>
                </S.Td>
                <S.Td>
                  <Link to={detailLink}>
                    {getListValueByKey(priority, incident.priority && incident.priority.priority)}
                  </Link>
                </S.Td>
                <S.Td>
                  <Link to={detailLink}>{incident.location && incident.location.address_text}</Link>
                </S.Td>
                {users && (
                  <S.Td>
                    <Link to={detailLink}>{getAssignedUserName(incident.assigned_user_id)}</Link>
                  </S.Td>
                )}
              </tr>
            );
          })}
        </tbody>
      </S.Table>
    </S.List>
  );
};

List.propTypes = {
  className: PropTypes.string,
  incidents: PropTypes.arrayOf(types.incidentType).isRequired,
  onChangeOrdering: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  priority: types.dataListType.isRequired,
  sort: PropTypes.string,
  stadsdeel: types.dataListType.isRequired,
  status: types.dataListType.isRequired,
};

export default List;
