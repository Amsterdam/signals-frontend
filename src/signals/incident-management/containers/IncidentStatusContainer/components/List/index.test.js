import React from 'react';
import { shallow } from 'enzyme';

import List from './index';

describe('<List />', () => {
  const statusList = [
    {
      key: '',
      value: 'Alle statussen'
    },
    {
      key: 'm',
      value: 'Gemeld'
    },
    {
      key: 'i',
      value: 'In afwachting van behandeling'
    },
    {
      key: 'b',
      value: 'In behandeling'
    },
    {
      key: 'o',
      value: 'Afgehandeld'
    },
    {
      key: 'h',
      value: 'On hold'
    },
    {
      key: 'a',
      value: 'Geannuleerd'
    }
  ];
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      id: '1',
      statusList,
      incidentStatusList: [
        {
          _links: { self: { href: '3' } },
          text: '',
          user: null,
          extern: false,
          state: 'm',
          created_at: '2018-07-13T11:48:45.555774+02:00'
        },
        {
          _links: { self: { href: '910' } },
          text: 'een tekst',
          user: 'opleiding@amsterdam.nl',
          extern: false,
          state: 'i',
          created_at: '2018-07-27T14:53:19.394361+02:00'
        },
        {
          _links: { self: { href: '911' } },
          text: 'is opgeruimd',
          user: '',
          state: 'o',
          created_at: '2018-07-27T14:53:35.989951+02:00'
        }
      ]
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    wrapper = shallow(
      <List {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render without list correctly', () => {
    props.incidentStatusList = [];
    wrapper = shallow(
      <List {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
