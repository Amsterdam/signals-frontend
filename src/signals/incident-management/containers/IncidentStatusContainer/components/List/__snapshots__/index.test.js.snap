// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render correctly 1`] = `
<div
  className="incident-status-container-list"
>
  <div
    className="incident-status-container-list__body"
  >
    <table
      cellPadding="0"
      cellSpacing="0"
      className="incident-status-container-list__body-list"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            Datum
          </th>
          <th
            className=""
          >
            Tijd
          </th>
          <th
            className=""
          >
            Status
          </th>
          <th
            className=""
          >
            Omschrijving
          </th>
          <th
            className=""
          >
            Gebruiker
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="3"
        >
          <td>
            13-07-2018
          </td>
          <td>
            11:48
          </td>
          <td />
          <td
            className="pre-wrap"
          />
          <td />
        </tr>
        <tr
          key="910"
        >
          <td>
            27-07-2018
          </td>
          <td>
            14:53
          </td>
          <td />
          <td
            className="pre-wrap"
          >
            een tekst
          </td>
          <td>
            opleiding@amsterdam.nl
          </td>
        </tr>
        <tr
          key="911"
        >
          <td>
            27-07-2018
          </td>
          <td>
            14:53
          </td>
          <td />
          <td
            className="pre-wrap"
          >
            is opgeruimd
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<List /> should render without list correctly 1`] = `
<div
  className="incident-status-container-list"
>
  <div
    className="incident-status-container-list__body"
  >
    <table
      cellPadding="0"
      cellSpacing="0"
      className="incident-status-container-list__body-list"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            Datum
          </th>
          <th
            className=""
          >
            Tijd
          </th>
          <th
            className=""
          >
            Status
          </th>
          <th
            className=""
          >
            Omschrijving
          </th>
          <th
            className=""
          >
            Gebruiker
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
